### ISSUE TITLE
Reverse-mode gradients through solve_sw are NaN whenever an aerosol_optics bundle is supplied

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-rrtmgp#30]
## Symptom

Differentiating `solve_sw` end to end produces non-finite cotangents as soon as **any**
`aerosol_optics` bundle is passed. Forward fluxes are finite; only the reverse pass is
affected. Without aerosol the same gradient is clean.

Observed on the RFMIP clear-sky profile for all of:

- `optical_depth = 0`
- `optical_depth = 1e-8`
- `optical_depth = 1e-2, ssa = 0.999` — a perfectly physical bundle, so this is not
  confined to a degenerate corner

Reproduce (differentiating w.r.t. temperature, `gpt_chunk=1` / unchunked):

```python
def loss(temp):
    fluxes = two_stream.solve_sw(
        p, temp, molecules, optics_lib, atmos_state, vmr_fields,
        aerosol_optics={
            'optical_depth': jnp.full(band_shape, 1e-2),
            'ssa': jnp.full(band_shape, 1.0),
            'asymmetry_factor': jnp.zeros(band_shape),
        },
        **cloud,
    )
    return jnp.sum(fluxes['flux_net'] ** 2)

jax.grad(loss)(temperature)   # -> NaN
```

## Scope

Pre-existing and independent of any current performance work: it reproduces on `02d3cc6`
(the `perf/two-stream-fusion-structure` head, which is PR #29 plus a fusion refactor) as
well as on the g-point-chunking branch where it was found. It was surfaced while writing a
gradient test for that branch, and that test deliberately excludes the aerosol path for
this reason — so the defect is currently untested rather than tracked by a failing test.

This matters downstream: jax-gcm differentiates through RRTMGP, and its aerosol paths
(MACv2-SP simple plumes, JAM modal aerosol) supply exactly this per-band bundle. Any
calibration or gradient-based workflow that includes aerosol radiative effects is blocked
by it.

## Where to look

`solve_sw` mixes the bundle at `rrtmgp/rte/two_stream.py:525` via
`optics_base.combine_optical_properties`, which normalises with
`jnp.maximum(..., _EPSILON)` guards:

```python
ssa_unnormalized = tau1 * ssa1 + tau2 * ssa2
g   = (tau1*ssa1*g1 + tau2*ssa2*g2) / jnp.maximum(ssa_unnormalized, self._EPSILON)
ssa = ssa_unnormalized / jnp.maximum(tau, self._EPSILON)
```

Those guards look adjoint-safe on inspection — `maximum` contributes a zero derivative on
the clamped branch — so the NaN has **not** been root-caused and may well originate
downstream of the mix rather than in it. Two candidates worth checking first:

1. **Conservative scattering reached via mixing.** An aerosol `ssa = 1` combined with a
   Rayleigh-dominated g-point can land the *combined* `ssa` at exactly 1.0 in float32,
   i.e. `k**2 = 0`. That is the failure class `8703bb5` fixed inside the two-stream by
   x²-parameterising `_sinhc2`/`_cosh2`; the question is whether the mixed path reaches a
   `sqrt`/division that guard does not cover. Note though that the `ssa = 0.999` case also
   NaNs, which argues against this being the whole story.
2. **A `0/0` in the asymmetry normalisation on the reverse path** when the scattering
   optical depth is ~0, where the forward value is clamped but the cotangent is not.

The established diagnosis technique for this class in this codebase is to differentiate the
**full solve** rather than kernels in isolation — the `8703bb5` NaN passed every standalone
kernel gradient test (their sweeps topped out at `ssa = 0.999`, never `1.0` exactly) and
only appeared end to end.

## Suggested acceptance

A gradient test that differentiates `solve_sw` end to end **with** an aerosol bundle, over
a sweep that includes `optical_depth = 0`, tiny-but-nonzero, and physical values, and with
`ssa = 1.0` included exactly rather than approached.
